python/cog/server/http.py

```diff
@@ -79,6 +81,44 @@ def create_app(
         input_type=InputType, output_type=OutputType
     )
 
+    if "train" in config:
```

Comment:
So the contract is "If a `cog.yaml` file has a `train` key, then the resulting OpenAPI schema has `TrainingInput` and `TrainingOutput` components defined; otherwise, those components aren't defined."

Reply:
I find it a bit weird that the prediction API uses the train one when run in predict mode, so you end up with two APIs that have the same interface.

Is this essentially for backwards compatibility?